### PR TITLE
[SYCL] Fix assert when calling get_binaries() on host

### PR DESCRIPTION
**Pull request by alexanderfle**
Signed-off-by: Alexander Flegontov <alexander.flegontov@intel.com>

### CODE REVIEW

File: sycl/source/detail/program_impl.cpp [hunk 1]
```diff
@@ -305,8 +305,8 @@ kernel program_impl::get_kernel(string_class KernelName,
 vector_class<vector_class<char>> program_impl::get_binaries() const {
   throw_if_state_is(program_state::none);
   vector_class<vector_class<char>> Result;
-  const detail::plugin &Plugin = getPlugin();
   if (!is_host()) {
+    const detail::plugin &Plugin = getPlugin();
```
Comment (bader) [inline sycl/source/detail/program_impl.cpp:309]:
```suggestion
  if (is_host())
    return {};
    
  vector_class<vector_class<char>> Result;
  const detail::plugin &Plugin = getPlugin();
```

Comment (bader) [reply sycl/source/detail/program_impl.cpp]: NOTE: you need remove closing } and fix formatting.

Comment (alexanderfle) [reply sycl/source/detail/program_impl.cpp]: thanks

File: sycl/source/detail/program_impl.cpp [hunk 2]
```diff
@@ -304,23 +304,25 @@ kernel program_impl::get_kernel(string_class KernelName,
 
 vector_class<vector_class<char>> program_impl::get_binaries() const {
   throw_if_state_is(program_state::none);
+  vector_class<vector_class<char>> Result;
```
Comment (bader) [inline sycl/source/detail/program_impl.cpp:307]:
```suggestion
```
This variable is re-defined below.

Comment (alexanderfle) [reply sycl/source/detail/program_impl.cpp]: thanks, I fixed it.

**Comment by alexanderfle** [comment]
@v-klochkov , Could you please approve the PR, again?

**Pull request merged**